controlplane/rosa/controllers/rosacontrolplane_controller.go

```diff
@@ -274,9 +274,15 @@ func (r *ROSAControlPlaneReconciler) reconcileNormal(ctx context.Context, rosaSc
 		ExpirationTimestamp(time.Now().Add(1 * time.Hour)).
 		Hypershift(cmv1.NewHypershift().Enabled(true))
 
+	// Set Netowrk
 	networkBuilder := cmv1.NewNetwork()
 	networkBuilder = networkBuilder.Type("OVNKubernetes")
-	networkBuilder = networkBuilder.MachineCIDR(*rosaScope.ControlPlane.Spec.MachineCIDR)
+	if rosaScope.ControlPlane.Spec.Network != nil {
+		networkBuilder = networkBuilder.MachineCIDR(*rosaScope.ControlPlane.Spec.Network.MachineCIDR)
```

Comment:
If this remains as a pointer, we need to check for `nil` before de-referencing it.

Reply:
will remove the pointer